lib/elixir/lib/string_io.ex

```diff
@@ -263,37 +257,21 @@ defmodule StringIO do
 
   ## get_line
 
-  defp get_line(encoding, prompt,
-                %{input: input, output: output, capture_prompt: capture_prompt} = s) do
-    case :unicode.characters_to_list(input, encoding) do
-      {:error, _, _} ->
-        {{:error, :collect_line}, s}
-      {:incomplete, _, _} ->
-        {{:error, :collect_line}, s}
-      chars ->
-        {result, input} = do_get_line(chars, encoding)
+  defp get_line(encoding, prompt, %{input: input} = s) do
+    with {result, input} <- binary_collect(input, encoding, "") do
+      result = case result do
+        "" -> :eof
+        r -> r
+      end
 
-        s =
-          if capture_prompt do
-            %{s | output: <<output::binary, IO.chardata_to_string(prompt)::binary>>}
-          else
-            s
-          end
+      s = capture_prompt(s, prompt)
 
-        {result, %{s | input: input}}
+      {result, %{s | input: input}}
```

Comment:
PS.: if it's because of naming purposes, you can always rename the function right?